### ISSUE TITLE
It seems that the craft-cookies plugin isn't able to set a cookie in Craft CMS v.3.5.17.1

### ISSUE THREAD

**Issue by dominikkrulak** [nystudio107/craft3-cookies#19]
### Describe the bug
I've recently upgraded Craft CMS 2 to the latest Craft CMS version and the craft-cookies plugin won't set a cookie.
I used your plugin in v.2 and it worked great.

In template I've tried all 3 non-secure methods to set a cookie and for the note, cookies are enabled in my browser.

### To reproduce
Steps to reproduce the behaviour:
1. Followed the upgrade instructions
2. Installed the plugin from the Plugin store
3. Noticed the problem

### Versions
- Plugin version: 1.1.12
- Craft version: 3.5.17.1

### Update
Firefox browser logs this warning massage
> Cookie “name” has been rejected because a non-HTTPS cookie can’t be set as “secure”.

which tells that setting a cookie in non-secure method like e.g.
```
{% do craft.cookies.set('name', 'value') %}
```
sets the cookie as secure.

Setting a cookie using a secure method like e.g.
```
{% do craft.cookies.setSecure('name', 'value') %}
```
works.

Something is not right because in the same environment (same PHP server config, same web server config, same browser config) worked all well under Craft CMS 2.

**Comment by khalwat** [comment]
Not sure offhand, perhaps it's due to changes in Yii2 under the hood?

**Comment by khalwat** [comment]
I bet this is why: https://craftcms.com/docs/3.x/config/config-settings.html#usesecurecookies

> Whether Craft will set the “secure” flag when saving cookies when using Craft::cookieConfig() to create a cookie.

> Valid values are true, false, and 'auto'. Defaults to 'auto', which will set the secure flag if the page you’re currently accessing is over https://. true will always set the flag, regardless of protocol and false will never automatically set the flag.

**Comment by dominikkrulak** [comment]
> will never automatically set the flag

This part of the quote from Craft CMS's docs tells me that it's up to me what type of cookie I set. Secure or unsecure regarding of protocol.

Let me explain it on examples.

Does `'useSecureCookies' => false,` mean that setting ~~an unsecure~~ a secure cookie 
e.g. `{% do craft.cookies.setSecure('name', 'value' %}`
should set the cookie (~~without~~ with "secure" flag if HTTPS protocol is present. If HTTPS is no present than without "secure" flag)?

And contrary, setting an unsecure cookie
e.g. `{% do craft.cookies.set('name', 'value' %}`
should set the cookie (without "secure" flag even if HTTPS protocol is present)?
 
Long story short, It's a manual work on the front-end than. 
Meaning that I control in my templates what type of cookie (secure or not secure) I set.


Well if I'm correct How I understand it than it still won't work.

**Comment by khalwat** [comment]
I'm afraid I don't really understand your question @dominikkrulak ?

My guess was that `'useSecureCookies' => false` in the Craft CMS `config/general.php` should cause it to not use secure cookies, and then yes, the Cookies plugin should then "just work"

Is that incorrect?

**Comment by dominikkrulak** [comment]
Yes @khalwat it is incorrect.
I'm in my local environment loading site with HTTP protocol and setting
`{% do craft.cookies.set('name', 'value' %}` doesn't set the cookie with `'useSecureCookies' => false,` set.

Sorry for having hard times around my question but I had it wrong. I've updated it. 
And I think it's straight forward simple now meaning   `'useSecureCookies' => false,` will only set unsecure cookie.
It's just that quote from docs

> will never automatically set the flag

confuses me.
